compiler/daml-lf-reader/src/DA/Daml/LF/Reader.hs

```diff
@@ -111,13 +120,27 @@ readDalfManifest dar = do
         lookup attrName attrs
     missingAttr attrName = Left $ "No " <> BSUTF8.toString attrName <> " attribute in manifest."
 
-readDalfs :: Archive -> Either String Dalfs
+readDalfs :: Archive -> Either String (Dalfs BSL.ByteString)
 readDalfs dar = do
     DalfManifest{..} <- readDalfManifest dar
     mainDalf <- getEntry dar mainDalfPath
     dalfs <- mapM (getEntry dar) (delete mainDalfPath dalfPaths)
     pure $ Dalfs mainDalf dalfs
 
+extractNameAndPackageIdFromPath :: FilePath -> (T.Text, LF.PackageId)
+extractNameAndPackageIdFromPath = second LF.PackageId . T.breakOnEnd "-" . T.pack . takeBaseName
```

Comment:
I'm sorry I forgot to mention, `fst . T.breakOnEnd "-"` will keep a trailing hyphen so we need something like this

```suggestion
extractNameAndPackageIdFromPath :: FilePath -> Maybe (T.Text, LF.PackageId)
extractNameAndPackageIdFromPath = bimapM (T.stripSuffix "-") (pure . LF.PackageId) . T.breakOnEnd "-" . T.pack . takeBaseName
```

One benefit here is that it forces us to handle the `Nothing` case, where the basename breaks our assumptions and doesn't actually contain a hyphen

Reply:
Oh i saw this in the docs and just completely forgot when implementing, cheers.